Bottom: a68c181b0cb559844ee84c750277438bdbd7bf5c
Top:    931edab48525dc659b78519fb0cc63fee9469fab
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-20 17:36:50 +0100

Refresh of tidy-up-pad-drawing-routines

---

diff --git a/src/crosshair.c b/src/crosshair.c
index 540cc1b..a9ed879 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -264,7 +264,7 @@ XORDrawElement (ElementTypePtr Element, LocationType DX, LocationType DY)
 
     if (PCB->InvisibleObjectsOn ||
         (TEST_FLAG (ONSOLDERFLAG, &moved_pad) != 0) == Settings.ShowSolderSide)
-      thindraw_pcb_pad (Crosshair.GC, &moved_pad, false, false);
+      gui->thindraw_pcb_pad (Crosshair.GC, &moved_pad, false, false);
   }
   END_LOOP;
   /* mark */
diff --git a/src/global.h b/src/global.h
index e449a34..0eca83b 100644
--- a/src/global.h
+++ b/src/global.h
@@ -54,6 +54,7 @@
 /* Forward declarations for structures the HIDs need.  */
 typedef struct BoxType BoxType, *BoxTypePtr;
 typedef struct polygon_st PolygonType, *PolygonTypePtr;
+typedef struct pad_st PadType, *PadTypePtr;
 typedef struct drc_violation_st DrcViolationType, *DrcViolationTypePtr;
 typedef struct rtree rtree_t;
 typedef struct AttributeListType AttributeListType, *AttributeListTypePtr;
@@ -310,14 +311,14 @@ typedef struct			/* a rat-line */
   Cardinal group1, group2;	/* the layer group each point is on */
 } RatType, *RatTypePtr;
 
-typedef struct			/* a SMD pad */
+struct pad_st			/* a SMD pad */
 {
   ANYLINEFIELDS;
   BDimension Mask;
   char *Name, *Number;		/* 'Line' */
   void *Element;
   void *Spare;
-} PadType, *PadTypePtr;
+};
 
 typedef struct
 {
diff --git a/src/hid.h b/src/hid.h
index b13e75b..d6f494a 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -382,6 +382,8 @@ typedef enum
                               const BoxType *clip_box);
     void (*thindraw_pcb_polygon) (hidGC gc_, PolygonType *poly,
                                   const BoxType *clip_box);
+    void (*fill_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
+    void (*thindraw_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
     void (*fill_rect) (hidGC gc_, int x1_, int y1_, int x2_, int y2_);
 
 
diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index b1b2123..9de587f 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -202,15 +202,15 @@ common_thindraw_pcb_polygon (hidGC gc, PolygonType *poly,
 void
 common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 {
-  int w = clear ? (mask ? Pad->Mask
-                        : Pad->Thickness + Pad->Clearance)
-                : Pad->Thickness;
+  int w = clear ? (mask ? pad->Mask
+                        : pad->Thickness + pad->Clearance)
+                : pad->Thickness;
   int x1, y1, x2, y2;
   int t = w / 2;
-  x1 = Pad->Point1.X;
-  y1 = Pad->Point1.Y;
-  x2 = Pad->Point2.X;
-  y2 = Pad->Point2.Y;
+  x1 = pad->Point1.X;
+  y1 = pad->Point1.Y;
+  x2 = pad->Point2.X;
+  y2 = pad->Point2.Y;
   if (x1 > x2 || y1 > y2)
     {
       int temp_x = x1;
@@ -220,7 +220,7 @@ common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
     }
   gui->set_line_cap (gc, Round_Cap);
   gui->set_line_width (gc, 0);
-  if (TEST_FLAG (SQUAREFLAG, Pad))
+  if (TEST_FLAG (SQUAREFLAG, pad))
     {
       /* slanted square pad */
       float tx, ty, theta;
@@ -271,70 +271,35 @@ common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 void
 common_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask)
 {
-  int w = clear ? (mask ? Pad->Mask
-                        : Pad->Thickness + Pad->Clearance)
-                : Pad->Thickness;
+  int w = clear ? (mask ? pad->Mask
+                        : pad->Thickness + pad->Clearance)
+                : pad->Thickness;
 
-  if (Pad->Point1.X == Pad->Point2.X &&
-      Pad->Point1.Y == Pad->Point2.Y)
+  if (pad->Point1.X == pad->Point2.X &&
+      pad->Point1.Y == pad->Point2.Y)
     {
-      if (TEST_FLAG (SQUAREFLAG, Pad))
+      if (TEST_FLAG (SQUAREFLAG, pad))
         {
           int l, r, t, b;
-          l = Pad->Point1.X - w / 2;
-          b = Pad->Point1.Y - w / 2;
+          l = pad->Point1.X - w / 2;
+          b = pad->Point1.Y - w / 2;
           r = l + w;
           t = b + w;
           gui->fill_rect (gc, l, b, r, t);
         }
       else
         {
-          gui->fill_circle (gc, Pad->Point1.X, Pad->Point1.Y, w / 2);
+          gui->fill_circle (gc, pad->Point1.X, pad->Point1.Y, w / 2);
         }
     }
   else
     {
-      gui->set_line_cap (gc, TEST_FLAG (SQUAREFLAG, Pad) ?
+      gui->set_line_cap (gc, TEST_FLAG (SQUAREFLAG, pad) ?
                                Square_Cap : Round_Cap);
       gui->set_line_width (gc, w);
 
-      gui->draw_line (gc, Pad->Point1.X, Pad->Point1.Y,
-                          Pad->Point2.X, Pad->Point2.Y);
-    }
-}
-
-void
-common_fill_pcb_pad (hidGC gc, PadTypePtr Pad, bool clear, bool mask)
-{
-  int w = clear ? (mask ? Pad->Mask
-                        : Pad->Thickness + Pad->Clearance)
-                : Pad->Thickness;
-
-  if (Pad->Point1.X == Pad->Point2.X &&
-      Pad->Point1.Y == Pad->Point2.Y)
-    {
-      if (TEST_FLAG (SQUAREFLAG, Pad))
-        {
-          int l, r, t, b;
-          l = Pad->Point1.X - w / 2;
-          b = Pad->Point1.Y - w / 2;
-          r = l + w;
-          t = b + w;
-          gui->fill_rect (gc, l, b, r, t);
-        }
-      else
-        {
-          gui->fill_circle (gc, Pad->Point1.X, Pad->Point1.Y, w / 2);
-        }
-    }
-  else
-    {
-      gui->set_line_cap (gc, TEST_FLAG (SQUAREFLAG, Pad) ?
-                               Square_Cap : Round_Cap);
-      gui->set_line_width (gc, w);
-
-      gui->draw_line (gc, Pad->Point1.X, Pad->Point1.Y,
-                          Pad->Point2.X, Pad->Point2.Y);
+      gui->draw_line (gc, pad->Point1.X, pad->Point1.Y,
+                          pad->Point2.X, pad->Point2.Y);
     }
 }
 
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index c844aff..5ac46ea 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -181,7 +181,7 @@ hid_extents_init (void)
 
   memset (&extents_hid, 0, sizeof (HID));
 
-  common_draw_helpers_init (&extentes_hid);
+  common_draw_helpers_init (&extents_hid);
 
   extents_hid.struct_size         = sizeof (HID);
   extents_hid.name                = "extents-extents";
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index f7e9e5d..b6d26f8 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -449,6 +449,8 @@ HID hid_nogui = {
   nogui_fill_polygon,
   nogui_fill_pcb_polygon,
   0 /* nogui_thindraw_pcb_polygon */ ,
+  nogui_fill_pcb_pad,
+  nogui_thindraw_pcb_pad,
   nogui_fill_rect,
   nogui_calibrate,
   nogui_shift_is_pressed,
@@ -504,6 +506,8 @@ apply_default_hid (HID * d, HID * s)
   AD (fill_polygon);
   AD (fill_pcb_polygon);
   AD (thindraw_pcb_polygon);
+  AD (fill_pcb_pad);
+  AD (thindraw_pcb_pad);
   AD (calibrate);
   AD (shift_is_pressed);
   AD (control_is_pressed);
